src/openassetio-core/tests/utils/PrintableTest.cpp

```diff
@@ -0,0 +1,191 @@
+    traitsData->setTraitProperty("a.long.namespaced.trait.that.goes.on.and.on.and.on",
+                                 "aStringTraitProperty", openassetio::Str("a string"));
+    traitsData->setTraitProperty("a.long.namespaced.trait.that.goes.on.and.on.and.on",
+                                 "aBoolTraitProperty", true);
+    traitsData->addTrait("a.trait.with.no.properties");
+
+    WHEN("An InfoDictionary is printed") {
+      THEN("It can be printed via fmt and <<") {
+        checkBasicPrintableByCharacters(infoDictionary, "{`key2`: False, `key1`: 'value1'}");
+      }
+    }
+
+    WHEN("A TraitSet is printed") {
+      THEN("It can be printed via fmt and <<") {
+        checkBasicPrintableByCharacters(traitSet, "{'trait2', 'trait1'}");
+      }
+    }
+
+    WHEN("TraitSets are printed") {
+      THEN("They can be printed via fmt and <<") {
+        checkBasicPrintableByCharacters(traitSets, "[{'trait2', 'trait1'}, {'trait4', 'trait3'}]");
+      }
+    }
+
+    WHEN("A BatchElementError is printed") {
+      THEN("It can be printed via fmt and <<") {
+        checkBasicPrintable(batchElementError, "invalidTraitSet: Invalid trait set");
+      }
+    }
+
+    WHEN("An EntityReference is printed") {
+      THEN("It can be printed via fmt and <<") {
+        checkBasicPrintable(entityReference, "test:///an_entity_reference");
+      }
+    }
+
+    WHEN("EntityReferences are printed") {
+      THEN("They can be printed via fmt and <<") {
+        checkBasicPrintable(entityReferences,
+                            "['test:///an_entity_reference_1', 'test:///an_entity_reference_2']");
+      }
+    }
+
+    WHEN("An Identifier is printed") {
+      THEN("It can be printed via fmt and <<") {
+        checkBasicPrintable(identifier, "an identifier");
+      }
+    }
+
+    WHEN("A Str is printed") {
+      THEN("It can be printed via fmt and <<") { checkBasicPrintable(str, "example string"); }
+    }
+
+    WHEN("A StrMap is printed") {
+      THEN("It can be printed via fmt and <<") {
+        checkBasicPrintableByCharacters(strMap, "{'key2': 'value2', 'key1': 'value1'}");
+      }
+    }
+
+    WHEN("A Capability is printed") {
+      THEN("It can be printed via fmt and <<") { checkBasicPrintable(capability, "publishing"); }
+    }
+
+    WHEN("A Context is printed") {
+      THEN("It can be printed via fmt and <<") {
+        // No closing brace on purpose, to account for variant managerState memory address
+        checkBasicPrintableContains(*context,
+                                    R"({'locale': {'aTrait': {'aIntTraitProperty': 2}}, )"
+                                    R"('managerState': 0x)");
+        checkBasicPrintableContains(context, R"('locale': {'aTrait': {'aIntTraitProperty': 2}}, )"
+                                             R"('managerState': 0x)");
+      }
+    }
+
+    WHEN("A TraitsData is printed") {
+      THEN("It can be printed via fmt and <<") {
```

Comment:
[fix] Test for the nullptr case

Reply:
https://github.com/OpenAssetIO/OpenAssetIO/pull/1329/commits/9b17e8a9244375e32257dc979fa75bd46c0e89f7